scripts/base/misc/version.zeek

```diff
@@ -49,39 +56,67 @@ export {
 
 function parse(version_string: string): VersionDescription
 	{
-	if ( /[[:digit:]]\.[[:digit:]][[:digit:]]?(\.[[:digit:]][[:digit:]]?)?(\-beta[[:digit:]]?)?(-[[:digit:]]+)?(\-debug)?/ != version_string )
+	if ( /[0-9]+\.[0-9]+(\.[0-9]+)?(-(beta|rc|dev)[0-9]*)?((-|\.)[0-9]+)?(-debug)?/ != version_string )
```

Comment:
Out of curiosity - is there a reason you prefer [0-9] to [[:digit:]]?

Reply:
Nothing that important: it's shorter and it's more explicit.  (Over)Using char class aliases may increase uncertainty: "does it really include the characters I think it does?" and takes some extra brain cells to remember "was it called 'digit' or 'number' ... 'alpha' or 'letter'".  So for smallish patterns, I don't find the extra indirection helpful.

Reply:
Fair enough :)